src/main/java/com/google/alpollo/SongDataBase.java

```diff
@@ -34,9 +34,10 @@ public static void saveAnalysisInfo(AnalysisInfo info) {
     List<SongCounter> songCounters = topSongs();
     // Song that we already have in database and it is on the last place in the top of songs
     SongCounter lessSearchedSongFromTop = songCounters.get(songCounters.size() - 1);
-
-    if (lessSearchedSongFromTop.getSearchCounter() < newSong.getSearchCounter()) {
+    
+    if (lessSearchedSongFromTop.getSearchCounter() < newSong.getSearchCounter() || songCounters.size() <= TOP_SIZE) {
       OfyService.ofy().delete().type(AnalysisInfo.class).id(lessSearchedSongFromTop.getSong().id());
+      info.setId();
```

Comment:
Why is that necessary? It looks like the id is set at the creation of the AnalysisInfo object